### DISCUSSION TITLE
Is it possible to optionnaly add $ cost in Proxy response json, in addition to token usage ?

### DISCUSSION THREAD

**Discussion by olad32** [BerriAI/litellm#3982] [Q&A]
proxy response could be : 
```
{
...
"usage": {
    "completion_tokens": 163,
    "prompt_tokens": 14684,
    "total_tokens": 14847,
    "cost": 1.123 <-- new
   }
}
```

This may be controlled by a new proxy config parameter

**Comment by ishaan-jaff** [comment]
@olad32 we could return it in the response headers @olad32. would that work ?

**Comment by olad32** [comment]
Yes thats also ok. Else it would have break openai response json format ? 

Le 3 juin 2024 16:51:07 GMT+02:00, Ishaan Jaff ***@***.***> a écrit :

**Reply by ghost** [reply]
yep exactly

**Reply by ghost** [reply]
we could do 

`x-litellm-response-cost: <calculated-cost>`

**Reply by ghost** [reply]
Curious, how would you use this?

**Reply by olad32** [reply]
hi, this will allows us to display cost usage per request to the end user

**Reply by ghost** [reply]
got it - we can do this, when do you need this by?

**Reply by olad32** [reply]
hello, we need this as soon as possible

**Reply by ghost** [reply]
got it - will pick this up today. 

Can we setup a direct support channel? @olad32 

Can do linkedin/discord/slack 

will help prioritize any issues your team has
